# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PWindowGlobal.ipdl
# Commit: 6709a3bc1a1c
# Full Hash: 6709a3bc1a1c4ac18dce0f6db78efe12b3e1f178
# Author: Kashav Madan <kmadan@mozilla.com>
# Date: 2019-07-15 21:39:46
# Regressor Bug: 1543251
# File Overlap Count: 1
# Description:
#   Bug 1543251 - Move hasBeforeUnload from PBrowser to PWindowGlobal, r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D37003
# ==============================================================================

diff -r 83881815f007 -r 6709a3bc1a1c dom/ipc/PWindowGlobal.ipdl
--- a/dom/ipc/PWindowGlobal.ipdl	Mon Jul 15 11:01:04 2019 +0000
+++ b/dom/ipc/PWindowGlobal.ipdl	Mon Jul 15 17:30:26 2019 +0000
@@ -60,6 +60,10 @@
   /// Send down initial document bit to the parent.
   async SetIsInitialDocument(bool aIsInitialDocument);
 
+  /// Tell the parent if this WindowGlobal has any "beforeunload" event
+  /// listeners.
+  async SetHasBeforeUnload(bool aHasBeforeUnload);
+
   /// Notify the parent that this PWindowGlobal is now the current global.
   async BecomeCurrentWindowGlobal();
 